### PR TITLE
Macvlan internal network should not change default gateway

### LINKED ISSUE #2406: Internal macvlan network overrides default gateway
I have container which needs to be connected to multiple overlay networks and one internal macvlan network.
Every time I connect the container to the macvlan network, it's default gateway will be set to non-existent default gateway.

I don't think internal networks should touch default gateway at all.

### PR DESCRIPTION
**Pull request by lemrouch**
Since docker container can be connected to combination of several
internal and external networks change of default gateway of the internal
ones breaks communication via the external ones.

This fixes only macvlan network type

Signed-off-by: Pavel Matěja <pavel@verotel.cz>

This should close #2406

### CODE REVIEW

**Comment by lemrouch** [comment]
What can I do to get this commit merged?

File: drivers/macvlan/macvlan_joinleave.go [hunk 1]
```diff
@@ -39,38 +39,49 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		return fmt.Errorf("could not find endpoint with id %s", eid)
 	}
 	// parse and match the endpoint address with the available v4 subnets
-	if len(n.config.Ipv4Subnets) > 0 {
-		s := n.getSubnetforIPv4(ep.addr)
-		if s == nil {
-			return fmt.Errorf("could not find a valid ipv4 subnet for endpoint %s", eid)
+	if !n.config.Internal {
+		if len(n.config.Ipv4Subnets) > 0 {
```
Comment (arkodg) [inline drivers/macvlan/macvlan_joinleave.go]: the log messages below can by de-duplicated by writing
if len(n.config.Ipv4Subnets) > 0 && !n.config.Internal with a comment saying why we are skipping this
same goes for Ipv6

Comment (lemrouch) [reply drivers/macvlan/macvlan_joinleave.go]: Those log messages are not duplicates. The later one is missing _Gateway: %s_ part.

**Comment by arkodg** [comment]
can you please make similar changes in https://github.com/docker/libnetwork/blob/6659f7f4d8c1ec7a412a33f8973423ad38d22982/drivers/ipvlan/ipvlan_joinleave.go#L53

**Comment by lemrouch** [comment]
Done, can you check it pls?

**Comment by arkodg** [comment]
will containers connected/attached to a single `internal` network (mtacvlan/overlay) in a swarm be able to communicate to each other ? (maybe they will since the the bridge connected to the veth might be on the same subnet, this should be tested and verified though :) )

**Comment by lemrouch** [comment]
Yes it does work. Network example is taken from https://github.com/docker/libnetwork/issues/2418#issuecomment-589097794
Run container on 1st node and check it's ip:
```
root@apex1:~# docker run -it --rm --name pg_client --network db_net postgres:11 /bin/bash
root@apex1:~# nsenter --net=/proc/`docker inspect --format '{{.State.Pid}}' pg_client`/ns/net ip a s
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
25: eth0@if3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:0a:40:31:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.64.49.2/24 brd 10.64.49.255 scope global eth0
       valid_lft forever preferred_lft forever
```
Run container on 2nd node and try to ping container on 1st node:
```
root@apex2:~# docker run -it --rm --name pg_client --network db_net postgres:11 /bin/bash
root@apex2:~# nsenter --net=/proc/`docker inspect --format '{{.State.Pid}}' pg_client`/ns/net ping -c 1 10.64.49.2
PING 10.64.49.2 (10.64.49.2) 56(84) bytes of data.
64 bytes from 10.64.49.2: icmp_seq=1 ttl=64 time=0.484 ms

--- 10.64.49.2 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.484/0.484/0.484/0.000 ms
```

**Pull request merged**